### PR TITLE
Receiving boxes

### PR DESCRIPTION
**Pull request by grischa**
Storage box mechanisms for ingesting files from a temporary holding area.

Usage:
Create a storage box with attribute `type: receiving` and foreign key `master_box` pointing to a permanent holding area.
Set up a celery beat schedule that runs the ingestion task `tardis_portal.ingest_received_files` periodically.

Would this do the job, @wettenhj ?

### CODE REVIEW

**Comment by wettenhj** [comment]
Thanks Grischa,

That definitely looks like it will do the job of moving (or copying) the
data and clearly annotating permanent, master and receiving storage boxes.

So now I guess the API needs to be updated so that when Datafile records
are created in this way:
http://mytardis.readthedocs.org/en/latest/api.html#via-staging-location, it
knows to create a "receiving" storage box?

Cheers,
James

**Comment by grischa** [comment]
Ah, yes. That is incomplete. There is already a `DEFAULT_RECEIVING_DIR` setting that is supposed to point to the default receiving location, but I don't think I'm auto-creating or guessing a storage box for it.

Alternatively, each MyData installation can get its own receiving box/location on setup and use the API to just register the path it pushes the file to. I.e. MyData controls its receiving area.
I'll think more about this and add to this PR

File: tardis/tardis_portal/migrations/0038_master_storage_boxes.py [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+        'tardis_portal.userprofile': {
+            'Meta': {'object_name': 'UserProfile'},
+            u'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
+            'isDjangoAccount': ('django.db.models.fields.BooleanField', [], {'default': 'True'}),
+            'rapidConnectEduPersonTargetedID': ('django.db.models.fields.CharField', [], {'max_length': '400', 'null': 'True', 'blank': 'True'}),
+            'user': ('django.db.models.fields.related.ForeignKey', [], {'to': u"orm['auth.User']", 'unique': 'True'})
+        }
+    }
+
+    complete_apps = ['tardis_portal']
```
Comment (grischa) [inline tardis/tardis_portal/migrations/0038_master_storage_boxes.py]: Codacy Issue found: [Final newline missing](https://www.codacy.com/app/grischa/mytardis/file/485553516/issues/source?bid=2125231&fileBranchId=2246351#l320)

Comment (grischa) [inline tardis/tardis_portal/migrations/0038_master_storage_boxes.py]: Codacy Issue found: [Final newline missing](https://www.codacy.com/app/grischa/mytardis/file/486073296/issues/source?bid=2125231&fileBranchId=2246351#l320)

**Pull request merged**